wandb/sdk/data_types.py

```diff
@@ -857,19 +859,30 @@ def seq_to_json(
 
 class Molecule(BatchableMedia):
     """
-    Wandb class for Molecular data
+    Wandb class for 3D Molecular data
 
     Arguments:
-        data_or_path: (string, io)
-            Molecule can be initialized from a file name or an io object.
+        data_or_path: (string, io, rdkit molecule)
+            Molecule can be initialized from a file name, an io object,
+            a SMILES string, or an rdkit.Chem.rdchem.Mol object.
     """
 
-    SUPPORTED_TYPES = set(
-        ["pdb", "pqr", "mmcif", "mcif", "cif", "sdf", "sd", "gro", "mol2", "mmtf"]
-    )
+    SUPPORTED_TYPES = {
+        "pdb",
+        "pqr",
+        "mmcif",
+        "mcif",
+        "cif",
+        "sdf",
+        "sd",
+        "gro",
+        "mol2",
+        "mmtf",
+        "mol",
+    }
     _log_type = "molecule-file"
 
-    def __init__(self, data_or_path: Union[str, "TextIO"], **kwargs: str) -> None:
+    def __init__(self, data_or_path: Union[str, "TextIO"], **kwargs) -> None:
```

Comment:
I recognize that this pattern of `data_or_path` exists for other data_types, but IMO, it is an anti-pattern and we should avoid doing more of it. Personally, I would prefer an alternative constructor that does not rely on type inference to dispatch the instantiation. Mainly, this is because the overloading of the constructor (or any method for that matter) becomes increasingly challenging to maintain over time as different parameter patterns and order of such become contradictory. Moreover, the `kwargs` here is not typechecked. 

As an alternative, consider making a constructor like `wandb.Molecule.from_rdkit(...)`. This would allow us to control the evolution of that API more precisely.

Reply:
Ah great idea Tim, I like this suggestion for the api!

Reply:
Great idea indeed! Implemented.